styleguide/screens/Patterns/Animation/Animation.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import dedent from 'dedent';
+
+import Specimen from '../../../components/Specimen/Specimen';
+import { D, H, T, C } from '../../../components/Scaffold/Scaffold';
+
+import Btn from '../../../../components/Btn/Btn';
+import Roll from '../../../../components/Animate/Roll';
+
+import m from '../../../../globals/modifiers.css';
+
+export default class Animation extends Component {
+  state = {
+    roll: false,
+  };
+
+  toggleRoll = () => {
+    this.setState(({ roll }) => ({
+      roll: !roll,
+    }));
+  };
+
+  render() {
+    const { roll } = this.state;
+
+    return (
+      <div>
+        <H level={ 1 }>Animation</H>
+        <D>
+          <H level={ 2 }>Roll</H>
+          <T elm="p" className={ m.mtr }>
+            <C>Roll</C> allows us to transition between two elements, pushing one
+            out, while pushing another in. This can be useful for highlighting a
+            change of state, where the resulting interface change is not obvious.
+          </T>
+          <Specimen
+            classNames={ {
+              root: m.mtr,
+              specimenContainer: m.par,
+            } }
+            code={ dedent`
+              <Btn onClick={ this.toggleRoll }>
+                <Roll width="10rem">
+                  { this.state.showMap
+                    ? <span id="list">Show list</span>
+                    : <span id="map">Show map</span>
+                  }
+                </Roll>
+              </Btn>
+            ` }
+          >
+            <Btn onClick={ this.toggleRoll }>
+              <Roll width="10rem">
```

Comment:
Do you think this could be a constant?

Reply:
@farazcsk What's the thinking behind that? I'm not sure it matters/is complex enough to warrant it

Reply:
@rdjpalmer I am probably wrong, but I just think it makes the example code snippet look clearner. Saying that, I can see how it would also maybe make it more confusing. I'll leave that up to you